extension/source/background/AggregatorController.ts

```diff
@@ -38,8 +36,10 @@ export default class AggregatorController {
   ) {}
 
   rpc = ensureType<PartialRpcImpl>()({
-    eth_sendTransaction: async (request) => {
-      const { providerId, params } = request;
+    eth_sendTransaction: async ({ providerId, params, origin }) => {
+      if (origin !== window.location.origin) {
```

Comment:
We don't require approval inside the extension.

Reply:
Also, `BlsWalletWrapper.connect` is slow (about 2 seconds I think), so putting this earlier means the user doesn't wait for those two seconds before seeing the approval dialog which is a much nicer experience.